pkg/scheduler/plugins/predicates/predicates.go

```diff
@@ -313,18 +313,44 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 		return nil
 	})
 
-	ssn.AddPredicateFn(pp.Name(), func(task *api.TaskInfo, node *api.NodeInfo) error {
+	ssn.AddPredicateResourceFn(pp.Name(), func(task *api.TaskInfo, node *api.NodeInfo) error {
 		nodeInfo, found := nodeMap[node.Name]
 		if !found {
-			return fmt.Errorf("failed to predicates, node info for %s not found", node.Name)
+			return fmt.Errorf("failed to predicate resource, node info for %s not found", node.Name)
+		}
+
+		if ok, reason := task.InitResreq.LessEqualWithReason(node.FutureIdle(), api.Zero); !ok {
+			klog.V(4).Infof("NodeFutureIdle resource predicates Task <%s/%s> on Node <%s> failed",
+				task.Namespace, task.Name, node.Name)
+			return api.NewFitError(task, node, reason)
 		}
 
 		if node.Allocatable.MaxTaskNum <= len(nodeInfo.Pods) {
-			klog.V(4).Infof("NodePodNumber predicates Task <%s/%s> on Node <%s> failed",
+			klog.V(4).Infof("NodePodNumber resource predicates Task <%s/%s> on Node <%s> failed",
 				task.Namespace, task.Name, node.Name)
 			return api.NewFitError(task, node, api.NodePodNumberExceeded)
 		}
 
+		if gpuDevice, ok := node.Others[api.GPUSharingDevice].(api.Devices); ok {
```

Comment:
Whether a type assertion is required here？

Reply:
We have [this issue](https://github.com/volcano-sh/volcano/issues/2738) to address it

Reply:
Yes, assertion judgment is required here, the current PR code has already been processed, I should comment on the wrong PR.